### PR TITLE
fix istioctl install display error

### PR DESCRIPTION
**Pull request by hanxiaop**
This PR is to fix issue #33857. It will show `This will install the Istio 1.10.1 default profile with` instead of `This will install the Istio 1.10.1  profile with` if no arg is provided.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[x] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: releasenotes/notes/33864.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+  - 33857
+releaseNotes:
+  - |
+    **Fixed** a message format bug for `istioctl install` command with the default file setting.
```
Comment (shamsher31) [inline releasenotes/notes/33864.yaml:8]:
```suggestion
    **Fixed** an issue where the `default` profile name was missing in the `istioctl install` confirmation prompt message.
```

**Review by shamsher31 — COMMENTED** [review]
LGTM 
```
$ istioctl install
This will install the Istio 1.11.0 default profile with ["Istio core" "Istiod" "Ingress gateways"] components into the cluster. Proceed? (y/N)
```

```
$ istioctl install --set profile=demo
This will install the Istio 1.11.0 demo profile with ["Istio core" "Istiod" "Ingress gateways" "Egress gateways"] components into the cluster. Proceed? (y/N)
```

```
$ istioctl install --set profile=minimal
This will install the Istio 1.11.0 minimal profile with ["Istio core" "Istiod"] components into the cluster. Proceed? (y/N)
```

**Review by shamsher31 — APPROVED** [review]

File: releasenotes/notes/33864.yaml (hunk 1 above)
Comment (monsteredc) [reply releasenotes/notes/33864.yaml]: @shamsher31 Thanks for the suggestion!

File: operator/cmd/mesh/install.go [hunk 2]
```diff
@@ -347,7 +347,14 @@ func getProfileNSAndEnabledComponents(iop *v1alpha12.IstioOperator) (string, str
 	}
 
 	if configuredNamespace := v1alpha12.Namespace(iop.Spec); configuredNamespace != "" {
-		return iop.Spec.Profile, configuredNamespace, enabledComponents, nil
+		return renderWithDefault(iop.Spec.Profile, "default"), configuredNamespace, enabledComponents, nil
```
Comment (howardjohn) [inline operator/cmd/mesh/install.go:350]: Should we just be setting the iop.Spec.Profile value somewhere? Not sure if we are already providing defaults directly to the spec in other parts of the code or not

Comment (monsteredc) [reply operator/cmd/mesh/install.go]: @howardjohn It seems like when flags are setting like "profile=demo", `overlaySetFlagValues` will set the path like {spec.profile: "demo"}. However, if none profile flag is provided, the profile field will be empty. I've revised and PTAL.

File: operator/pkg/manifest/shared.go [hunk 3]
```diff
@@ -537,6 +539,20 @@ func overlaySetFlagValues(iopYAML string, setFlags []string) (string, error) {
 	return string(out), nil
 }
 
+// addDefaultProfileFlagIfNotSet adds the `profile=default` value to set flags if no profile value is set.
+func addDefaultProfileFlagIfNotSet(setFlags []string) []string {
```
Comment (howardjohn) [inline operator/pkg/manifest/shared.go:543]: I don't think this is the right layer to apply this fix. What about something more like
```diff
--- a/operator/pkg/manifest/shared.go
+++ b/operator/pkg/manifest/shared.go
@@ -212,6 +212,9 @@ func GenIOPFromProfile(profileOrPath, fileOverlayYAML string, setFlags []string,
        if ns := GetValueForSetFlag(setFlags, "values.global.istioNamespace"); ns != "" {
                finalIOP.Namespace = ns
        }
+       if finalIOP.Spec.Profile == "" {
+               finalIOP.Spec.Profile = name.DefaultProfileName
+       }
        return util.ToYAMLWithJSONPB(finalIOP), finalIOP, nil
 }
```

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: operator/pkg/manifest/shared.go (hunk 3 above)
Comment (monsteredc) [reply operator/pkg/manifest/shared.go]: @howardjohn Thanks! Looks much better.

**Review by howardjohn — APPROVED** [review]
Thanks!

**Pull request merged**